[cmake] Rework FindJsonSchemaBuilder on the model of FindTextureBuilder

marmeladema opened the pull request:
## Description
It allows to specify the location of the JsonSchemaBuilder
binary with the WITH_JSONSCHEMABUILDER cmake variable.
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
JsonSchemaBuilder and TextureBuilder are currently the two binaries that need to be previously compiled when cross compiling Kodi because they are not available from outside of project.
It ease the cross-compilation of Kodi by allowing to specify the directory where JsonSchemaBuilder is located on the host.

## How Has This Been Tested?
It has been tested on my Linux machine, but definitely needs testing on other systems, especially Windows

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

wsnipex commented on cmake/modules/FindJsonSchemaBuilder.cmake:
```diff
@@ -1,26 +1,41 @@
 #.rst:
 # FindJsonSchemaBuilder
-# ---------------------
+# -----------------
```
please remove this change

wsnipex commented on cmake/modules/FindJsonSchemaBuilder.cmake:
```diff
@@ -1,26 +1,41 @@
+                                                       IMPORTED_LOCATION "${DEPENDENCIES_DIR}/bin/json-rpc/JsonSchemaBuilder")
+  else()
+    if(WITH_JSONSCHEMABUILDER)
+      get_filename_component(_tppath ${WITH_JSONSCHEMABUILDER} ABSOLUTE)
```
please use a unique var here, _tppath is copied from texturepacker

wsnipex reviewed (commented):
thanks.
only 2 minors, otherwise looks fine

marmeladema committed:
[cmake] Rework FindJsonSchemaBuilder on the model of FindTextureBuilder

It allows the specify the location of the JsonSchemaBuilder
binary with the WITH_JSONSCHEMABUILDER cmake variable.

marmeladema commented:
I did the requested changes. Let me know if i have to do anything else.

emanuel4you commented:
| [ 1%] Generating ServiceDescription.h
| [ 1%] Built target wrapper
| cd /home/emanuel/OE2.5/opendreambox/build/dm920/tmp-glibc/work/cortexa15hf-neon-vfpv4-oe-linux-gnueabi/kodi/18.0+gitrAUTOINC+8b68a90123-r0/build/build && JsonSchemaBuilder::JsonSchemaBuilder /home/emanuel/OE2.5/opendreambox/build/dm920/tmp-glibc/work/cortexa15hf-neon-vfpv4-oe-linux-gnueabi/kodi/18.0+gitrAUTOINC+8b68a90123-r0/git/xbmc/interfaces/json-rpc/schema/version.txt /home/emanuel/OE2.5/opendreambox/build/dm920/tmp-glibc/work/cortexa15hf-neon-vfpv4-oe-linux-gnueabi/kodi/18.0+gitrAUTOINC+8b68a90123-r0/git/xbmc/interfaces/json-rpc/schema/license.txt /home/emanuel/OE2.5/opendreambox/build/dm920/tmp-glibc/work/cortexa15hf-neon-vfpv4-oe-linux-gnueabi/kodi/18.0+gitrAUTOINC+8b68a90123-r0/git/xbmc/interfaces/json-rpc/schema/methods.json /home/emanuel/OE2.5/opendreambox/build/dm920/tmp-glibc/work/cortexa15hf-neon-vfpv4-oe-linux-gnueabi/kodi/18.0+gitrAUTOINC+8b68a90123-r0/git/xbmc/interfaces/json-rpc/schema/types.json /home/emanuel/OE2.5/opendreambox/build/dm920/tmp-glibc/work/cortexa15hf-neon-vfpv4-oe-linux-gnueabi/kodi/18.0+gitrAUTOINC+8b68a90123-r0/git/xbmc/interfaces/json-rpc/schema/notifications.json
| make -f build/cores/dll-loader/exports/CMakeFiles/wrapper.def.dir/build.make build/cores/dll-loader/exports/CMakeFiles/wrapper.def.dir/depend
| /bin/sh: 1: JsonSchemaBuilder::JsonSchemaBuilder: not found
| build/interfaces/json-rpc/schema/CMakeFiles/generate_json_header.dir/build.make:68: recipe for target 'build/ServiceDescription.h' failed
| make[2]: *** [build/ServiceDescription.h] Error 127
| make[2]: Leaving directory '/home/emanuel/OE2.5/opendreambox/build/dm920/tmp-glibc/work/cortexa15hf-neon-vfpv4-oe-linux-gnueabi/kodi/18.0+gitrAUTOINC+8b68a90123-r0/build'
| CMakeFiles/Makefile2:3338: recipe for target 'build/interfaces/json-rpc/schema/CMakeFiles/generate_json_header.dir/all' failed
| make[1]: *** [build/interfaces/json-rpc/schema/CMakeFiles/generate_json_header.dir/all] Error 2
| make[1]: *** Waiting for unfinished jobs....

I've got that build.make error new?
I've fixed it by swaping the "JsonSchemaBuilder::JsonSchemaBuilder" to the real path.

DaveTBlake commented:
>It has been tested on my Linux machine, but definitely needs testing on other systems, **especially Windows**

I take it no one actutally did this before merge, or pinged anyone to do it. It breaks JSON on Windows. But sorry I don't see how to fix other than revert

emanuel4you commented:
still the same error

emanuel4you commented:
-DWITH_JSONSCHEMABUILDER=${STAGING_BINDIR_NATIVE}/JsonSchemaBuilder \

works for me now setting jsb path explicitly openembedded's EXTRA_OECMAKE var
